esp-hal/src/aes/mod.rs

```diff
@@ -444,12 +438,11 @@ pub mod dma {
             }
             self.enable_dma(true);
             self.enable_interrupt();
-            self.set_mode(mode);
+            self.aes.write_mode(mode);
             self.set_cipher_mode(cipher_mode);
             self.write_key(key.into());
 
-            // TODO: verify 16?
-            self.set_num_block(16);
```

Comment:
Was this ever working?

Rounding up here seems incorrect but can't think of an alternative.